<a href="https://colab.research.google.com/github/govindkoli27/heights-of-male-and-female-bu-countries-analysis/blob/main/Height_of_male_and_female.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd 
import numpy as np
import matplotlib.pyplot as plt 
import seaborn as sns 
import plotly.express as px
import matplotlib.pyplot as plt
%matplotlib inline


In [5]:
height= pd.read_csv('Height of Male and Female by Country 2022.csv')

In [6]:
height

Rank            Country Name  Male Height in Cm  Female Height in Cm  \
0       1             Netherlands             183.78               170.36   
1       2              Montenegro             183.30               169.96   
2       3                 Estonia             182.79               168.66   
3       4  Bosnia and Herzegovina             182.47               167.47   
4       5                 Iceland             182.10               168.91   
..    ...                     ...                ...                  ...   
194   195              Mozambique             164.30               155.42   
195   196        Papua New Guinea             163.10               156.89   
196   197         Solomon Islands             163.07               156.79   
197   198                    Laos             162.78               153.10   
198   199             Timor-Leste             160.13               152.71   

     Male Height in Ft  Female Height in Ft  
0                 6.03                 5.59  
1                 6.01                 5.58  
2                 6.00                 5.53  
3                 5.99                 5.49  
4                 5.97                 5.54  
..                 ...                  ...  
194               5.39                 5.10  
195               5.35                 5.15  
196               5.35                 5.14  
197               5.34                 5.02  
198               5.25                 5.01  

[199 rows x 6 columns]

In [7]:
height['Mean Height']= (height['Male Height in Cm'] + height['Female Height in Cm'])/2

Added a mean height column for each country

In [8]:
height.drop(["Rank",'Male Height in Ft','Female Height in Ft'], axis=1)

Country Name  Male Height in Cm  Female Height in Cm  \
0               Netherlands             183.78               170.36   
1                Montenegro             183.30               169.96   
2                   Estonia             182.79               168.66   
3    Bosnia and Herzegovina             182.47               167.47   
4                   Iceland             182.10               168.91   
..                      ...                ...                  ...   
194              Mozambique             164.30               155.42   
195        Papua New Guinea             163.10               156.89   
196         Solomon Islands             163.07               156.79   
197                    Laos             162.78               153.10   
198             Timor-Leste             160.13               152.71   

     Mean Height  
0        177.070  
1        176.630  
2        175.725  
3        174.970  
4        175.505  
..           ...  
194      159.860  
195      159.995  
196      159.930  
197      157.940  
198      156.420  

[199 rows x 4 columns]

In [9]:
height.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 199 entries, 0 to 198
Data columns (total 7 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   Rank                 199 non-null    int64  
 1   Country Name         199 non-null    object 
 2   Male Height in Cm    199 non-null    float64
 3   Female Height in Cm  199 non-null    float64
 4   Male Height in Ft    199 non-null    float64
 5   Female Height in Ft  199 non-null    float64
 6   Mean Height          199 non-null    float64
dtypes: float64(5), int64(1), object(1)
memory usage: 11.0+ KB


In [10]:
height.isnull().sum()

Rank                   0
Country Name           0
Male Height in Cm      0
Female Height in Cm    0
Male Height in Ft      0
Female Height in Ft    0
Mean Height            0
dtype: int64

In [11]:
height.describe()

Rank  Male Height in Cm  Female Height in Cm  Male Height in Ft  \
count  199.000000         199.000000           199.000000         199.000000   
mean   100.000000         173.089045           160.942915           5.678794   
std     57.590508           4.949832             4.076377           0.162510   
min      1.000000         160.130000           150.910000           5.250000   
25%     50.500000         169.490000           158.240000           5.560000   
50%    100.000000         173.530000           160.620000           5.690000   
75%    149.500000         176.510000           163.870000           5.790000   
max    199.000000         183.780000           170.360000           6.030000   

       Female Height in Ft  Mean Height  
count           199.000000   199.000000  
mean              5.280402   167.015980  
std               0.133870     4.432794  
min               4.950000   156.420000  
25%               5.190000   163.892500  
50%               5.270000   166.995000  
75%               5.375000   170.355000  
max               5.590000   177.070000

In [12]:
fig = px.bar(height, x="Country Name", y="Male Height in Cm", color="Male Height in Cm", title="Male Height by Country (cm)")
fig.show()

We can know the male height from all the countries where dark blue tells us the shortest height and yellow tells us about highest male height from different countries.

In [13]:
fig = px.bar(height, x="Country Name", y="Female Height in Cm", color="Female Height in Cm", title="Female Height by Country (cm)")
fig.show()

We can know the female height from all the countries where dark blue tells us the shortest height and yellow tells us about highest female height from different countries.

In [14]:
fig = px.histogram(height, x="Male Height in Cm",title='Male Height (cm) Histogram',color_discrete_sequence=['indianred'])
fig.show()

We can see the number of countries whose avg. male height is between 165 cm to 170 cm and very few countries have male whose height more than 180 cm or below 160 cm.

In [15]:
fig = px.histogram(height, x="Female Height in Cm",title='Female Height (cm) Histogram',
                    color_discrete_sequence=['indianred'])
fig.show()


We can see the number of countries whose avg. female height is between 157 cm to 165 cm and very few countries have female whose height more than 167 cm or below 155 cm.

In [16]:
highest_male = height.sort_values(by='Male Height in Cm', ascending=False).head(10)[['Country Name','Male Height in Cm']]
highest_female = height.sort_values(by='Female Height in Cm', ascending=False).head(10)[['Country Name','Female Height in Cm']]



shortest_male = height.sort_values(by='Male Height in Cm', ascending=False).tail(10)[['Country Name','Male Height in Cm']]

shortest_female = height.sort_values(by='Female Height in Cm', ascending=False).tail(10)[['Country Name','Female Height in Cm']]


In [17]:
fig = px.bar(highest_male, x='Country Name', y='Male Height in Cm', title='Top 10 Countries with highest male')
fig.show()

Top 10 countries names with highest male height is between 180 cm  to 185 cm.

In [18]:
fig = px.bar(highest_female, x='Country Name', y='Female Height in Cm', title='Top 10 Countries with highest female')
fig.show()

Top 10 countries names with highest female height is between 167 cm  to 171 cm.

In [19]:
fig = px.pie(shortest_female, values='Female Height in Cm', names='Country Name')
fig.show()

Top 10 countries names with shortest female height is between 150 cm  to 155 cm.

In [20]:
fig = px.pie(shortest_male, values='Male Height in Cm', names='Country Name')
fig.show()

Top 10 countries names with shortest male height is between 160 cm  to 165 cm.